rs/boundary_node/ic_boundary/src/bouncer/firewall.rs

```diff
@@ -107,6 +107,7 @@ impl Set {
             .map(|x| Expression::String(x.to_string()))
             .collect::<Vec<_>>();
 
+        #[allow(clippy::clone_on_copy)]
```

Comment:
nit: I am surprised this needs to be in this MR for CI to pass

Reply:
[Here](https://dfinity.slack.com/archives/CL7Q2RXUM/p1724161764044049) are more details

Reply:
@stedfn can you add a note in the code about this?